supporting_components/graph.py

```diff
@@ -440,18 +440,21 @@ def copy(self):
         copy = Graph(self.directed)
         copy.max_colornum = self.max_colornum
         vertices_old_to_new = {}
+        colors = {}
 
         for v in self.vertices:
             vertices_old_to_new[v] = Vertex(copy)
             vertices_old_to_new[v].label = v.label
             vertices_old_to_new[v].colornum = v.colornum
             vertices_old_to_new[v].graph_label = v.graph_label
             vertices_old_to_new[v].degree_fixed = v.degree_fixed
+            colors.setdefault(v.colornum, list()).append(v)
```

Comment:
En als deze colors dictionary aan de copy graph toegevoegd wordt, zouden de vertices in die graph dan ook niet de nieuwe zijn? Dus inplaats van ```append(v)``` -> ```append(vertices_old_to_new[v])```

Reply:
Daar heb je gelijk aan, maar deze methode wordt niet meer halverwege gebruikt, dus dan maakt het niet uit. Maar heb het veranderd.